### ISSUE TITLE
Finetuning code of anomaly detection

### ISSUE THREAD

**Issue by Jubi-Hwang** [moment-timeseries-foundation-model/moment#50]
Can you provide the full finetuning code of anomaly detection tutorial?
I want to know the processing of “original” in your code when dealing with the tiltAPB2 dataset.

# Compute loss
loss = criterion(output.reconstruction, original)

Also, I have a question about the shape of GT labels when using more than one ‘c_channels’.
When we train or finetuning the model, what is the dimension of “labels”? (batch_size, # of channels) or (batch_size, 1)?
I think in test mode the latter is right, in train mode the former is right. Is it correct?

Thank you.

**Comment by mononitogoswami** [comment]
Hi Jubi, 
The full fine-tuning code can be found in the `MOMENT-Research` repository: https://github.com/moment-timeseries-foundation-model/moment-research/blob/main/scripts/finetuning/anomaly_detection.py

During fine-tuning, we fine-tune MOMENT on an unsupervised reconstruction task i.e. we fine-tune the pre-training (reconstruction) head on in-domain time series, without any labels. 

We only use the labels for evaluation. 

If you want to do supervised fine-tuning, then you might have to tweak the fine-tuning code suit your problem setting. 

Let me know if you have any more questions, and thanks for your interest in MOMENT! 

I'll close this issue for now, but don't hesitate to open it should you have further questions. 

Best,
Mononito
